storops/lib/job_helper.py

```diff
@@ -0,0 +1,97 @@
+# coding=utf-8
+# Copyright (c) 2017 Dell Inc. or its subsidiaries.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+from __future__ import unicode_literals
+
+import logging
+import threading
+import time
+from collections import OrderedDict
+
+from storops.lib.common import singleton
+from storops.unity.resource.job import UnityJobList, wait_job_completion
+
+log = logging.getLogger(__name__)
+
+
+@singleton
+class JobHelper(object):
+    def __init__(self, cli, interval=5):
+        self._cli = cli
+        self._interval = interval
+        self.jobs = OrderedDict()
+        self.started = False
+
+    def add_job(self, job):
+        if job.id not in self.jobs:
+            self.jobs[job.id] = job
+
+    def remove_job(self, job):
+        if job.id in self.jobs:
+            del self.jobs[job.id]
+
+    def get_job(self, job):
+        if job.id in self.jobs:
+            return self.jobs[job.id]
+
+    def wait_job(self, job, timeout, interval):
+        def update_func(job):
+            return self.get_job(job)
+
+        self.add_job(job)
+        job = wait_job_completion(job, timeout=timeout, interval=interval,
+                                  update_func=update_func)
+        self.remove_job(job)
```

Comment:
If timeout, the exception raised. `job` will not be removed nor returned.

Reply:
Fixed.